File: PP_02.01/DrivingSchoolWebAPI2/DrivingSchoolAPIModels/ApiModels/RegisterModel.cs
using System.ComponentModel.DataAnnotations;

namespace DrivingSchoolAPIModels
{
    /// <summary>
    /// Модель для регистрации нового пользователя
    /// </summary>
    public class RegisterModel : Person
    {
        /// <summary>
        /// Электронная почта
        /// </summary>
        [EmailAddress]
        [Required(ErrorMessage = "Требуется адрес электронной почты")]
        public string Email { get; set; }
        /// <summary>
        /// Пароль
        /// </summary
        [Required(ErrorMessage = "Требуется пароль")]
        public string Password { get; set; }
        /// <summary>
        /// Номер телефона
        /// </summary>
        [Phone]
        public string? PhoneNumber { get; set; }
        /// <summary>
        /// Имя
        /// </summary>
        [Required(ErrorMessage = "Требуется имя")]
        public string FirstName { get; set; }
        /// <summary>
        /// Фамилия
        /// </summary>
        [Required(ErrorMessage = "Требуется фамилия")]
        public string LastName { get; set; }

    }
}

File: PDP/DrivingSchoolAppAndroidClient/app/src/main/java/com/example/drivingschoolappandroidclient/api/models/RegisterModel.kt
package com.example.drivingschoolappandroidclient.api.models

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import java.io.ByteArrayOutputStream

data class RegisterModel(
    var email: String,
    var password: String,
    override var firstName: String,
    override var lastName: String,
    override var patronymic: String? = null,
    var phoneNumber: String?,
    override var profilePictureBytes: ByteArray?,

    ) : IPerson{

    private var _image: Bitmap? = null
    override var profilePicture: Bitmap?
        get() {
            if (_image == null){
                profilePictureBytes?.let{
                    _image = BitmapFactory.decodeByteArray(it,0,it.size)
                }
            }
            return _image
        }
        set(value) {
            _image = value
            value?.let{
                val os = ByteArrayOutputStream()
                it.compress(Bitmap.CompressFormat.JPEG,100,os)
                profilePictureBytes = os.toByteArray()
            }
        }
    }
